### PR TITLE
Adds rain, I guess

### PR DESCRIPTION
**Pull request by FlattestGuitar**
I still need sprites for the actual rain. The current "acid rain" stuff /tg/ has is a joke and looks ugly.

There's also a rarer subtype of it - the lovely thunderstorm, with small odds of giving you 75 burn damage and a pretty light to look at if you're the lucky type.

I experimented a bunch with different stuff, but it seems I'll need to get the sun working properly first in order to get some sort of a global darkness setting going - clouds and whatnot.

Also, this changes the areas of the planet to `area/planet/outdoors` where applicable. We'll be using those for every turf that isn't roofed because of the way the weather system works.

### CODE REVIEW

File: code/datums/weather/weather_types.dm [hunk 1]
```diff
@@ -221,3 +223,65 @@
 	var/resist = L.getarmor(null, "acid")
 	if(prob(max(0,100-resist)))
 		L.acid_act(20,20)
+
+/datum/weather/regular_rain
+	name = "rain"
+	desc = "Some stay dry and others feel the pain"
+
+	telegraph_duration = 400
+	telegraph_message = "<span class='notice'>You see clouds occlude the sky...</span>"
```
Comment (SamHPurp) [inline code/datums/weather/weather_types.dm]: `occlude` seems like an odd language choice to me

File: _maps/map_files/Persephone/Persephone.dmm [hunk 2]
```diff
@@ -1,138 +1,122 @@
...
+"bK" = (/obj/structure/flora/grass/jungle,/obj/structure/flora/grass/jungle,/obj/structure/flora/grass/jungle,/turf/open/floor/planet/grass,/area/planet/outdoors)
+"bL" = (/obj/structure/flora/ausbushes/fullgrass,/obj/structure/flora/ausbushes/brflowers,/obj/structure/flora/ausbushes/grassybush,/turf/open/floor/planet/grass,/area/planet/outdoors)
+"bM" = (/obj/structure/flora/grass/jungle/b,/obj/structure/flora/ausbushes/grassybush,/turf/open/floor/planet/grass,/area/planet/outdoors)
+"bN" = (/obj/structure/flora/ausbushes/ppflowers,/mob/living/simple_animal/butterfly,/turf/open/floor/planet/grass,/area/planet/outdoors)
+"bO" = (/turf/closed/wall/planet/wood_wall,/area/planet)
+"bP" = (/turf/open/floor/planet/wood,/area/planet)
+"bQ" = (/obj/structure/chair/wood,/turf/open/floor/planet/wood,/area/planet)
+"bR" = (/obj/structure/table/wood,/turf/open/floor/planet/wood,/area/planet)
+"bS" = (/obj/structure/table/wood,/obj/item/device/flashlight/lamp,/turf/open/floor/planet/wood,/area/planet)
+"bT" = (/obj/structure/closet/crate/wooden,/obj/item/device/flashlight/flare,/turf/open/floor/planet/wood,/area/planet)
```
Comment (SamHPurp) [inline _maps/map_files/Persephone/Persephone.dmm]: Why did you remove the area type for the village?

File: code/datums/weather/weather_types.dm [hunk 3]
```diff
@@ -221,3 +223,65 @@
...
+	if(prob(1) && prob(1))//0.01% chance every tick
+		var/list/possible_targets = list()
+		for(var/atom/A in view(L, 6))
+			if(istype(get_area(A), /area/planet/outdoors))
+				possible_targets += A
+		var/atom/target = pick(possible_targets)
+		var/turf/T = get_step(get_step(target, NORTH), NORTH)
+		T.Beam(target, icon_state="lightning[rand(1,12)]", time = 5)
+		playsound(T, 'sound/effects/thunder.ogg', 50, 1, -1)//sound effect by Mike Koenig. Thanks, Mike!
+		target.visible_message("<span class='danger'>[target] is hit by a lightning!</span>", "<span class='userdanger'>You're hit by a lightning!</span>")
```
Comment (Citinited) [inline code/datums/weather/weather_types.dm]: Should be `hit by lightning` or `struck by lightning`, not `hit by a lightning`

File: code/datums/weather/weather_types.dm [hunk 4]
```diff
@@ -221,3 +223,65 @@
...
+
+	probability = 10
+
+/datum/weather/regular_rain/thunderstorm/impact(mob/living/L)
+	..()
+	if(!L.client)//this whole thing will only try to strike stuff around mobs, for the sake of not being boring
+		return
+	if(prob(1) && prob(1))//0.01% chance every tick
+		var/list/possible_targets = list()
+		for(var/atom/A in view(L, 6))
```
Comment (Citinited) [inline code/datums/weather/weather_types.dm]: I'd probably do this as a turf check, then strike everything located on that turf. IMO it makes more sense to strike everything on  a turf (if lightning strikes directly next to you then you *should* fry), and should cut down on processing time when stuff does get hit by lightning.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 5]
```diff
@@ -214,12 +214,13 @@
...
 			msg += "<B>[t_He] [t_has] severe cellular damage.</B>\n"
 
-
-	if(fire_stacks > 0)
-		msg += "[t_He] [t_is] covered in something flammable.\n"
-	if(fire_stacks < 0)
-		msg += "[t_He] looks a little soaked.\n"
-
+	switch(fire_stacks)
+		if(-20 to -10)
```
Comment (Citinited) [inline code/modules/mob/living/carbon/human/examine.dm]: `if(-INFINITY to -10)`

File: code/datums/weather/weather_types.dm [hunk 6]
```diff
@@ -269,14 +269,14 @@
 		return
 	if(prob(1) && prob(1))//0.01% chance every tick
 		var/list/possible_targets = list()
-		for(var/atom/A in view(L, 6))
-			if(istype(get_area(A), /area/planet/outdoors))
-				possible_targets += A
+		for(var/turf/T in view(L, 6))
+			if(istype(get_area(T), /area/planet/outdoors))
+				possible_targets += T.contents//adds a list to a list, should be fine
```
Comment (Citinited) [inline code/datums/weather/weather_types.dm]: Eh I meant like "pick a turf, strike literally everything on that tile", but if you want to do it this way, that's fine too 😄

File: code/datums/weather/weather_types.dm [hunk 7]
```diff
@@ -221,3 +223,65 @@
...
+	telegraph_message = "<span class='notice'>You see a thunderstorm brewing in the distance...</span>"
+
+	telegraph_sound = 'sound/ambience/thunder_start.ogg'
+	weather_sound = 'sound/ambience/thunder_mid.ogg'
+
+	probability = 10
+
+/datum/weather/regular_rain/thunderstorm/impact(mob/living/L)
+	..()
+	if(!L.client)//this whole thing will only try to strike stuff around mobs, for the sake of not being boring
```
Comment (SamHPurp) [inline code/datums/weather/weather_types.dm]: Is that actually how this works in practice? It doesn't seem like it would unless there is an adjoining bit of code I'm not seeing?

Comment (FlattestGuitar) [reply code/datums/weather/weather_types.dm]: If the mob L doesn't have a client, it won't consider it. Only mobs with players in them will be taken into account. I'm not sure how smart of a solution it is, but it makes all lightnings hit people or around people.
